### ISSUE TITLE
content-encoding is accepted or returned via S3 API

### ISSUE THREAD

**Issue by sdether** [seaweedfs/seaweedfs#3984]
**Describe the bug**
The `content-encoding` header is not respected using the S3 API. No error is raised when the header is provided for `put-object` but if it is stored, it is not included in `get-object` or `head-object` response.

**System Setup**
Default docker container:
```
docker run -p 8333:8333 chrislusf/seaweedfs server -s3
```

**Expected behavior**
```
$ aws --endpoint-url http://localhost:8333 s3api put-object \
           --bucket testbucket \
           --key test.txt.gz \
           --body tmp/test.txt.gz \
           --content-encoding gzip
{
    "ETag": "\"d8e8fca2dc0f896fd7cb4cb0031ba249\""
}

$ aws --endpoint-url http://localhost:8333 s3api head-object --bucket testbucket --key tmp/test.txt.gz
{
    "AcceptRanges": "bytes",
    "LastModified": "2022-11-15T22:53:44+00:00",
    "ContentLength": 5,
    "ETag": "\"d8e8fca2dc0f896fd7cb4cb0031ba249\"",
    "ContentDisposition": "inline; filename=\"test.txt.gz\"",
    "ContentEncoding": "gzip",
    "ContentType": "application/x-gzip",
    "Metadata": {}
}
```
When executed against the above docker container the `ContentEncoding` entry is missing from the response.

**Comment by aronneagu** [comment]
Using the docker image `chrislusf/seaweedfs:3.35` I have managed to reproduce the bug, but building from source off master (git hash 2b910d1cf), it does seem to return ContentEncoding

```
# aws --endpoint-url http://localhost:8333 s3api head-object --bucket testbucket --key test.txt.gz
{
    "AcceptRanges": "bytes",
    "LastModified": "Sat, 26 Nov 2022 13:28:06 GMT",
    "ContentLength": 0,
    "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"",
    "ContentDisposition": "inline; filename=\"test.txt.gz\"",
    "ContentEncoding": "gzip",
    "ContentType": "application/gzip",
    "Metadata": {}
}
```

According to S3 specs, the header `ContentEncoding` should be `Content-Encoding` (similar to all headers starting with `Content`, so I am claiming there is still a bug

https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html

**Comment by sdether** [comment]
I believe `ContentEncoding` vs. `Content-Encoding` are representational differences between the raw HTTP response and how AWS-CLI formats it. I.e. see the CLI docs for head-object https://docs.aws.amazon.com/cli/latest/reference/s3api/head-object.html where all the headers that are usually written with dashes are turned into Pascal case.

Which means the issue ought to be just one of needing a docker container built off master rather than `3.35`

**Comment by aronneagu** [comment]
You are right, its a difference in how the response is formatted, when the metadata is returned in the body of the HTTP response vs just as HTTP headers (was clear from the aws docs on head-object)

Releases are created quite frequently, so a new one should be available soon, otherwise running `make` will easily compile and install the binary
